core/src/banking_stage/immutable_deserialized_packet.rs

```diff
@@ -98,6 +100,22 @@ impl ImmutableDeserializedPacket {
         self.compute_budget_details.clone()
     }
 
+    /// Returns true if the transaction's compute unit limit is at least as
+    /// large as the sum of the static builtins' costs.
+    /// This is a simple sanity check so the leader can discard transactions
+    /// which are statically known to exceed the compute budget, and will
+    /// result in no useful state-change.
+    pub fn compute_unit_limit_above_static_builtins(&self) -> bool {
```

Comment:
This could be part of `ImmutableDeserializedPacket::new() -> Result`, can drop packet there if not enough cu limit. Just to clarify that it is standalone function to support scheduler_controller?

Reply:
made it a separate function more just more easily document the behavior and use in filter call.
also didn't want to put it directly in `new` so we don't need to change any of the existing tests.